### PR TITLE
KVStore: Add some log for Learner Read

### PR DESCRIPTION
**Pull request by CalvinNeo**
### What problem does this PR solve?

Issue Number: ref #8845

Problem Summary:

Without this PR, we can't find correspondence between read index request and result.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

**Comment by JaySon-Huang** [comment]
Can you describe why it cause the problem?

File: dbms/src/Storages/KVStore/Read/ReadIndexDataNode.cpp [hunk 1]
```diff
@@ -82,37 +82,65 @@ void ReadIndexDataNode::runOneRound(const TiFlashRaftProxyHelper & helper, const
                 e.second->update(history_success_tasks->second);
             }
 
+            LOG_DEBUG(
+                DB::Logger::get(),
+                "[Learner Read] Read Index in Batch(use histroy), max_ts={} region_id={} waiting_tasks={} "
+                "running_tasks={}, histroy_ts={}",
```
Comment (JaySon-Huang) [inline dbms/src/Storages/KVStore/Read/ReadIndexDataNode.cpp:88]:
```suggestion
                "running_tasks={} histroy_ts={}",
```

**Comment by CalvinNeo** [comment]
> Can you describe why it cause the problem?

This is not a bugfix(which will come after this ). For root cause see the issue https://github.com/pingcap/tiflash/issues/8845.

File: dbms/src/Storages/KVStore/Read/ReadIndexDataNode.cpp [hunk 2]
```diff
@@ -82,37 +82,65 @@ void ReadIndexDataNode::runOneRound(const TiFlashRaftProxyHelper & helper, const
...
         }
         else
         {
             auto run_it = running_tasks.lower_bound(max_ts);
-            if (run_it == running_tasks.end())
+            bool should_build_running_task = run_it == running_tasks.end();
+            bool build_success = false;
+            if (should_build_running_task)
             {
+                // If we can't attach us to some running_tasks.
```
Comment (JaySon-Huang) [inline dbms/src/Storages/KVStore/Read/ReadIndexDataNode.cpp:104]:
```suggestion
                // If we can't attach to some running_tasks.
```

File: dbms/src/Storages/KVStore/Read/ReadIndexDataNode.cpp
Comment (JaySon-Huang) [inline dbms/src/Storages/KVStore/Read/ReadIndexDataNode.cpp:1]: Will these logging make logging flood in our testing env?

Comment (JaySon-Huang) [reply dbms/src/Storages/KVStore/Read/ReadIndexDataNode.cpp:1]: Seems the number of logging is `num of region * num of read request`

Comment (CalvinNeo) [reply dbms/src/Storages/KVStore/Read/ReadIndexDataNode.cpp:1]: Yes, it could, I will try to eliminate the most frequent ones

Comment (CalvinNeo) [reply dbms/src/Storages/KVStore/Read/ReadIndexDataNode.cpp:1]:
```
[root@tc-tiflash-0 logs]# cat server.log.2024-04-23-08:52:54.063 | grep "Read Index in Batch" | wc -l
625257
[root@tc-tiflash-0 logs]# cat server.log.2024-04-23-08:52:54.063 | grep "Reuse read result in cache" | wc -l
47
[root@tc-tiflash-0 logs]# cat server.log.2024-04-23-08:52:54.063 | grep "Start read index" | wc -l
49469
[root@tc-tiflash-0 logs]# cat server.log.2024-04-23-08:52:54.063 | grep "Learner Read" | wc -l
872643
```

Comment (CalvinNeo) [reply dbms/src/Storages/KVStore/Read/ReadIndexDataNode.cpp:1]: fixed by changing "Read Index in Batch" logs into trace level.

**Review by JaySon-Huang — APPROVED** [review]

**Pull request merged**